pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -134,10 +156,38 @@ func injectAutoInstrumentation(pod *corev1.Pod, _ string, _ dynamic.Interface) e
 	if len(libsToInject) == 0 {
 		return nil
 	}
+	// Inject env variables used for Onboarding KPIs propagation
+	if isApmInstrumentationEnabled(pod.Namespace) {
+		// if Single Step Instrumentation is enabled, inject DD_INSTRUMENTATION_INSTALL_TYPE:k8s_single_step
+		_ = injectEnv(pod, singleStepInstrumentationInstallTypeEnvVar)
+	} else {
+		// if local library injection is enabled, inject DD_INSTRUMENTATION_INSTALL_TYPE:k8s_lib_injection
+		_ = injectEnv(pod, localLibraryInstrumentationInstallTypeEnvVar)
+	}
 
 	return injectAutoInstruConfig(pod, libsToInject, autoDetected)
 }
 
+func injectApmTelemetryConfig(pod *corev1.Pod) {
+	// inject DD_INSTRUMENTATION_INSTALL_TIME with current Unix time
+	instrumentationInstallTime := os.Getenv(instrumentationInstallTimeEnvVarName)
+	if instrumentationInstallTime == "" {
+		instrumentationInstallTime = strconv.FormatInt(time.Now().Unix(), 10)
```

Comment:
This would set the install time to the time that the library is injected, this value should be computed when the Cluster Agent starts which is closer to the intent of using APM.